### ISSUE TITLE
Add run name, run description (& other top level experiment/run related metadata columns) to runs UI table & other run/experiment specific tables on Aim

### ISSUE THREAD

**Issue by ashutoshsaboo** [aimhubio/aim#1426]
## 🚀 Feature

1. Show run name, run description and other top level metadata related columns (if they are missing) in runs table to see all runs easily. Currently run timestamp is the only one that's shown, which doesn't provide any helpful/informative info to understand what that run represents other than clicking on it which is not great. If run name as well as run description is shown it can contain more informative info from the user and can be accordingly helpful. 

Other columns that can also be added to the runs table are here in screenshot below, please add all but please don't render them by default on the runs table as else it'll clutter the screen. This way, user can choose to check/uncheck these top-level metadata fields related to the experiment/run if they want to.: 
<img width="638" alt="Screen Shot 2022-03-01 at 16 15 47" src="https://user-images.githubusercontent.com/9900945/156167792-ed0d48ef-fd4d-4643-a3f6-12ccf7087e47.png">

2. The ability to save configuration columns pinned/unpinned/hidden etc in the Runs table - so each time user doesn't have to re-configure for the same in the runs ui page. 

### Motivation

See above 

### Pitch

See above 

### Alternatives

None

### Additional context

cc/ @roubkar @SGevorg @gorarakelyan

**Comment by roubkar** [comment]
Hey @ashutoshsaboo, thanks for opening this issue 🙌

**Comment by ashutoshsaboo** [comment]
If this doesn't already exist, then please add a small filtering/search function to filter/search for relevant columns easily. It'd be helpful to search and filter with a small search bar rather than manually scrolling. 

Can these columns (as toggle-able but not rendered by default) be added to the other explorer specific tables (and other similar tables across Aim too) as well, say image/text etc across Aim? @roubkar

**Comment by roubkar** [comment]
Re searching: you can search by column name in manage columns popover mid pane section and pin columns to left/right, reorder and toggle visibility(hide/show), will this resolve your issue?

<img width="750" alt="Screen Shot 2022-03-01 at 17 40 50" src="https://user-images.githubusercontent.com/9951653/156179502-40f124a4-5f3f-4009-b19d-3e93c964688a.png">

Re `Can these columns be added to the other explorer specific tables`: yes, once we will add those columns to Runs page it will be easy to add to the rest of the Explorer pages.

**Comment by ashutoshsaboo** [comment]
- Perfect, yes the search thing that you mentioned about works fine. 
- Thanks for the heads up on adding the these columns to other tables as well. It'd be great if we can track for adding these columns to other tables in this issue itself (especially given text explorer is in the works before this one). @roubkar

**Comment by ashutoshsaboo** [comment]
Currently fix that was merged in 3.8.0-dev3 release - the correct run name/description isn't being rendered in the runs table (see screenshots below). The popover in the last screenshot which shows names of all runs/experiments must also have run names rather than only run times which are shown currently - perhaps you could show `<run name> | <run time> | <run duration` - the latter 2 are already being shown but without run_name it's very tough to see from the popover menu which was the actual run that corresponds to it. 

From runs table:
![Screenshot 2022-03-16 at 6 21 13 PM](https://user-images.githubusercontent.com/9900945/158664877-3a193c98-cbbd-4a90-b795-7b374947af33.png)
Inside a completed run's page the header:
![Screenshot 2022-03-16 at 6 20 34 PM](https://user-images.githubusercontent.com/9900945/158664885-7a398be9-6d3a-4fcf-8a79-76a2c30f7b21.png)
Popover on clicking on the run's page at the top to navigate to other expts/runs:
![Screenshot 2022-03-17 at 12 16 04 AM](https://user-images.githubusercontent.com/9900945/158664917-d91cd92f-f62c-4414-a291-8ee8cfa735bd.png)


For reference, i'm logging run name/expt name/run description with this blurb below:
```
run = Run(log_system_params=True, experiment=args.aim_experiment_name)
run.name = args.aim_run_name
run.description = args.aim_run_description
```

@VkoHov @SGevorg : please take note.

**Comment by VkoHov** [comment]
Hey @ashutoshsaboo, thanks for this feedback.
We will add run names to popover in 3.8.0.

**Comment by mihran113** [comment]
hey @ashutoshsaboo! I've opened an issue and a PR for the run name/description not being set.
 https://github.com/aimhubio/aim/issues/1548
 This will be shipped with `v3.8.0`. Thanks a lot for the report. 🙌

**Comment by ashutoshsaboo** [comment]
Looks like the bug was that you weren't persisting it correctly if i understand it correctly, What about backfill for runs that have already been completed with this attribute, but it wasn't set for them due to the bug? @mihran113

**Comment by ashutoshsaboo** [comment]
@VkoHov @mihran113 : In the delete runs popup, the run time is shown but the run name isn't shown making it difficult to infer which run exactly is being deleted. Can both run name and run timestamp (which is already present along with expt name) in the same?

Also, I noticed for longer expt/run names - the delete runs popup doesn't show the full name (I know it's visible on hover). If possible, can there be some sort of text warping (i.e, have fixed width, but make height dynamic based on amount text content like Excel does) in the delete runs popup to see the full expt name/run name in the same along with the full text hover that is in place currently? Same is also happening in the experiments popover (screenshot that I had attached above), if possible please do it there too. 

The reasoning is simple: It makes it much easier to see full names in one go while scrolling rather than hovering over each one-by-one.

**Comment by VkoHov** [comment]
> @VkoHov @mihran113 : In the delete runs popup, the run time is shown but the run name isn't shown making it difficult to infer which run exactly is being deleted. Can both run name and run timestamp (which is already present along with expt name) in the same?
> 
> Also, I noticed for longer expt/run names - the delete runs popup doesn't show the full name (I know it's visible on hover). If possible, can there be some sort of text warping (i.e, have fixed width, but make height dynamic based on amount text content like Excel does) in the delete runs popup to see the full expt name/run name in the same along with the full text hover that is in place currently? Same is also happening in the experiments popover (screenshot that I had attached above), if possible please do it there too.
> 
> The reasoning is simple: It makes it much easier to see full names in one go while scrolling rather than hovering over each one-by-one.

@ashutoshsaboo I believe this is out of the scope of the original issue here. 
I am going to start working on it now, but this needs a separate issue. Please open it. 🙌

**Comment by ashutoshsaboo** [comment]
@VkoHov : there you go - https://github.com/aimhubio/aim/issues/1593 .

**Comment by VkoHov** [comment]
@ashutoshsaboo, I opened a separate issue regarding adding experiment-related metadata to explorers tables. Track the progress [here](https://github.com/aimhubio/aim/issues/1671).

Run name and description were added to tables and shipped with v3.8. Please try it out and let me know if it works as expected.
